core/src/main/java/org/pentaho/reporting/platform/plugin/ParameterXmlContentHandler.java

```diff
@@ -381,6 +381,16 @@ public void createParameterContent( final OutputStream outputStream,
         hideOutputParameterIfLocked( report, reportParameters );
       }
 
+      // Hide the parameter, like output-target, if report opened in PIR (editor/viewer) or report is prpt.
+      if ( ( ( path.equals( "/" ) || path.endsWith( ".prpti" ) ) && ( requestParams.getParameter( "json" ) != null ) )
+        || path.endsWith( "prpt" ) ) {
+        final ParameterDefinitionEntry definitionEntry = reportParameters.get( "includeReportSpecification" );
+        if ( definitionEntry instanceof AbstractParameter ) {
+          final AbstractParameter parameter = (AbstractParameter) definitionEntry;
+          parameter.setHidden( true );
+        }
+      }
```

Comment:
Why the need to specialize for `prpti` or `prpt`? Shouldn't the new `includeReportSpecification` parameter always be hidden, regardless?

Reply:
This is confusing to me. 
What exact case is this covering? 
What do you mean by "viewer via prpt"?
What previous code for `output-type` are you referring to?

Reply:
Got it, regarding viewing a `prpti` using the report viewer.

About the `output-type`, I couldn't find any caller of the `createParameterContent` method passing a `true` value to the `overrideOutputType` parameter. Regardless, at least, in this case, it would be the caller (report viewer, PIR, ,...) to decide and have the logic for when to hide or not. Could it be a parameter as well?

Reply:
I too couldn't find case where `overrideOutputType` is true.

 I see it is [hidden ](https://github.com/pentaho/pentaho-platform-plugin-reporting/blob/c08a99b4fb0d7f9be56fcd2833abd52a34a7e019/core/src/main/java/org/pentaho/reporting/platform/plugin/ParameterXmlContentHandler.java#L677) via attribute `LOCK_PREFERRED_OUTPUT_TYPE` . [PIR ](https://github.com/pentaho/pentaho-platform-plugin-interactive-reporting/blob/714eabedf12dac4a90fd5cef3e3c0589e7e18c29/core/src/main/java/com/pentaho/iadhoc/service/AdhocReportContentGenerator.java#L99-L101) hides it